subsys/net/lib/azure_iot_hub/Kconfig

```diff
@@ -5,18 +5,58 @@
 #
 
 module = AZURE_IOT_HUB
-module-prompt = "Azure IoT Hub [EXPERIMENTAL]"
+module-prompt = "Azure IoT Hub"
 orsource "../Kconfig.cloud_$(CLOUD_SERVICE_SELECTOR)"
 
 menuconfig $(module)
 	bool
-	prompt "Azure IoT Hub [EXPERIMENTAL]" if !CLOUD_SERVICE_MUTUAL_EXCLUSIVE
+	prompt "Azure IoT Hub" if !CLOUD_SERVICE_MUTUAL_EXCLUSIVE
+	select NET_SOCKETS_POSIX_NAMES
```

Comment:
This isn't the best choice, sooner or later someone complains that they want to use the library with `CONFIG_POSIX_API`, which is not compatible with this config. The library should either support both variants, or stick to native Zephyr APIs (`zsock_*`).

Reply:
Updated to support both variants